### ISSUE TITLE
To be or not to be?

### ISSUE THREAD

**Issue by vpeschenkov** [vpeschenkov/genosia#1]
@dkhamsing Hi there! I want to know your opinion about the project. By now, it can generate `README.MD`, `STORE.MD`, `ARCHIVE.MD` & `links` files. Also, it can validate the `categories`. What do you think about it? 

There is a built [app](https://github.com/vpeschenkov/genosia/releases/download/v0.0.1/genosia.zip) for testing

### The command to generate `*.MD` files
`genosia generate --content --path contents.json`
### The command to generate `links` files
`genosia generate --links --path contents.json`
### The command to validate `categories`
`genosia validate --content --path contents.json`

`--path` flag  is optional. If it is not set, `genosia` tries to find the `contents.json` in the same folder.
## Files
- There is a new [`osia_convert.rb`](https://github.com/vpeschenkov/genosia/tree/master/Sources/Core/Builder/Strategies)
- There is a new [`osia_validate_categories.rb`](https://github.com/vpeschenkov/genosia/blob/master/Sources/Core/Validator/CategoriesValidator.swift)

**Comment by dkhamsing** [comment]
This looks very interesting but we’ll have to explore how they work with our continuous integration

Thx for sharing 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by vpeschenkov** [comment]
@dkhamsing It works well, but, only on `osx`.

```swift
os: osx
language: ruby
rvm: 2.2
before_script:
  - curl -O -L https://github.com/vpeschenkov/Genosia/releases/download/v0.0.2/genosia-osx.zip && unzip genosia-osx.zip
  - gem install awesome_bot
  - gem install danger
script:
  - ./genosia generate --links
  - awesome_bot check-unique.txt --allow-ssl -a 302 -w xbmc/xbmc,Cherry/master,preyproject
  - awesome_bot check-links.txt  --allow-ssl -a 403 --allow-dupe -w wheelmap.org,sourceforge,radioparadise,iltofa,douban,preyproject,laravel-china
after_script:
  - awesome_bot check-info.txt -a 403
  - danger
```
I've tried to investigate how to compile it on `linux` but couldn't get worked executable :-(

**Comment by dkhamsing** [comment]
Wow cool

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by dkhamsing** [comment]
Hey Victor, while I appreciate your effort to build this tool, we are currently using Circle CI to generate the README https://github.com/dkhamsing/open-source-ios-apps/blob/master/circle.yml and it does not support `osx` for free 

Also I think I may prefer these silly ruby scripts right now.. they may be simpler to use and change

**Comment by vpeschenkov** [comment]
I agree with you. I've tried to make this work on Circle CI but without result. I've hoped to compile it for linux but swift package manger currently doesn't support static linking on it. However, It was interesting for me and now we know that to rewriting on swift is not only about rewriting . Also, I got another one argument what swift is not so good how the Apple says 😀.

Thank you for your time!
